psycopg/psycopg/waiting.py

```diff
@@ -198,14 +203,147 @@ def wakeup(state: Ready) -> None:
                     loop.remove_writer(fileno)
             fileno, s = gen.send(ready)
 
-    except TimeoutError:
+    except AsyncIOTimeoutError:
         raise e.ConnectionTimeout("connection timeout expired")
 
     except StopIteration as ex:
         rv: RV = ex.args[0] if ex.args else None
         return rv
 
 
+try:
```

Comment:
...same here: I would prefer to rely on the anyio objects implemented in separate modules. This woudn't even need conditional import, because it would be imported by `connection_anyio.py`